src/main/java/by/itechart/api/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package by.itechart.api.service.impl;
+
+import by.itechart.api.entity.User;
+import by.itechart.api.repository.UserRepository;
+import by.itechart.api.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public User create(User user) {
+        return userRepository.save(user);
+    }
+
+    @Override
+    public User update(User user) {
+        return userRepository.saveAndFlush(user);
+    }
+
+    @Override
+    public void delete(Long id) {
+        userRepository.deleteById(id);
```

Comment:
Since we have `deleted_at` column for tracking data of deletion, I think it would make more sense to implement `logical deletion` by just marking the entity as `deleted` instead of actually removing it from the table

Reply:
Can you provide me some articles about that?

Reply:
[Here](https://www.brentozar.com/archive/2020/02/what-are-soft-deletes-and-how-are-they-implemented/) you can find some introductory information about the concept